docs/.vuepress/public/demo.html

```diff
@@ -0,0 +1,28 @@
+<div>
```

Comment:
Why are you adding these files in the public folder of vuepress? 

Same goes for all the files and folders you added in the usage folder.

I think its better to just integrete these kind of examples in the docs pages itself (md files) and use place a chart chart custom chart undeaneath it with that config.

Maby we can look into the plugin to add them so we can disable the user input so the user only sees the chart

Reply:
@LeeLenaleee Cool question! So, these public files represent self-contained, pre-built Chart.js examples that I embed into the step-by-step guide. If you'd like to, consider running `pnpm docs:dev` and checking out http://localhost:8080/docs/master/getting-started/; you might enjoy it 😄

Some of these examples rely on an external dependency to fetch substantially sized "real-world data," and embedding the library in the docs would be unreasonable. I've explored embedding the datasets directly, but they are dozens of KBs, so it's also not an option.

> Maby we can look into the plugin to add them so we can disable the user input, so the user only sees the chart

Yep, it's a viable option for the file you're commenting (`demo.html`). I've submitted a PR to `vuepress-theme-chartjs` that would allow that: https://github.com/simonbrunel/vuepress-theme-chartjs/pull/4

> I think its better to just integrete these kind of examples in the docs pages itself (md files) 

Right. With the PR above, that one can be replaced with something like this:
```
```js chart-editor
const data = [
  { year: 2010, count: 10 },
  { year: 2011, count: 20 },
  { year: 2012, count: 15 },
  { year: 2013, count: 25 },
  { year: 2014, count: 22 },
  { year: 2015, count: 30 },
  { year: 2016, count: 28 },
];

module.exports = {
  config: {
    type: 'bar',
    data: {
      labels: ['Red', 'Blue', 'Yellow', 'Green', 'Purple', 'Orange'],
      datasets: [{
        label: '# of Votes',
        data: [12, 19, 3, 5, 2, 3],
        borderWidth: 1
      }]
    },
    options: {
      scales: {
        y: {
          beginAtZero: true
        }
      }
    }
  },
  chartOnly: true
};
```
```

Let's wait for @simonbrunel to merge the PR to `vuepress-theme-chartjs`. Alternatively, I can submit a follow-up PR later after the theme is updated so this one is not blocked.

Reply:
Let's wait for @simonbrunel to merge the PR to `vuepress-theme-chartjs`. Alternatively, I can submit a follow-up PR later after the theme is updated so this one is not blocked.

Reply:
While I appreciate the work I am against merging this for the following reasons:

1. I find it a security risk, those js files are big and hard to read so difficult to know if you putted something in there that's not supposed to be there.

2. Its a maintenance burden to update them Everytime something in the codebase has changed

3. Sample in the docs should be minimalistic in my eyes and not rely on external data which can change or go down. If we want an example with loads of data we should add it in the samples section with generated data

Reply:
I fully agree with @LeeLenaleee, especially on the maintenance part. There should be no pre-built files in the repository. I'm not even sure how the `usage/*` files are supposed to be generated? where are the sources? IMO, embedding complex examples in the docs should go through some services like CodePen or alternatives, so users have access to the entire source code.

But as a getting started, I would indeed expect something minimalistic as mentioned by @LeeLenaleee.

Reply:
@LeeLenaleee @simonbrunel This is excellent feedback; thanks! Honestly, I dislike the current state of things just like you do. It was an attempt to have [interactive charts rather than images](https://github.com/chartjs/Chart.js/pull/10816#discussion_r1001181459), but it looks ugly now. Let me roll this PR back to just images and remove all the bloat.

@simonbrunel has a great point on CodePen/CodeSandbox/etc. However, they would still be external to docs and could potentially break, so let's skip that options for now.

Reply:
Rolled back to images. (No new files under `.vuepress` anymore.) IMO, much better now. WDYT?

Reply:
Images make sense to me. A CodePen or similar could work, but the maintenance burden is there. Maybe there are ways we could automatically update the codepen during deploys?